Library/Homebrew/cli/named_args.rb

```diff
@@ -103,6 +103,20 @@ def to_formulae_to_casks(only: parent&.only_formula_or_cask, method: nil)
                                                 .map(&:freeze).freeze
       end
 
+      # Returns formulae and casks after validating that a tap is present for each of them.
+      def to_formulae_and_casks_with_taps
+        to_formulae_and_casks.each do |formula_or_cask|
+          case formula_or_cask
+          when Formula
+            odie "Formula #{formula_or_cask.name} is not in a tap!" unless formula_or_cask.tap
+          when Cask::Cask
+            odie "Cask #{formula_or_cask.token} is not in a tap!" unless formula_or_cask.tap
```

Comment:
Might be better to gather all the missing formulae/casks outside of the loop and print a single error message. This avoids having e.g. `brew audit badformula1 badcask1 badformula2 badcask2` printing 4 different error messages but instead one good one.